src/main/java/com/zaxxer/nuprocess/linux/ProcessEpoll.java

```diff
@@ -36,10 +39,19 @@
  */
 class ProcessEpoll extends BaseEventProcessor<LinuxProcess>
 {
+   private static final int EPOLL_MAX_EVENTS = 20;
    private final int epoll;
-   private final EpollEvent triggeredEvent;
+   private final EpollEvent[] triggeredEvents;
+   private final EpollEvent firstTriggeredEvent;
    private final List<LinuxProcess> deadPool;
 
+   // preferring (assuming uncontented) locking over memory allocations
+   // used in #registration and #queueWrite methods
+   private final EpollEvent tempEventForRegistration = new EpollEvent();
+   private final EpollEvent tempEventForQueueWrite = new EpollEvent();
+   private final Lock tempEventForRegistrationLock = new ReentrantLock();
+   private final Lock tempEventForQueueWriteLock = new ReentrantLock();
```

Comment:
Hm. I put in the comment that I prefer _assumed_ uncontented locking over jna native call to malloc. Indeed I am optimizing for my use case at the cost of yours.

Since we are going to malloc for the buffers per process anyway might as well for event per process.